### PR TITLE
Fractional reference bin indices for correcting TOF axis

### LINKED ISSUE #33222: Fractional reference bin indices for correcting TOF axis
**Is your feature request related to a problem? Please describe.**

Scientists responsible for ILL Time-of-flight instruments have reported recently that they are not able to ensure the elastic peak in their corrected measurements falls at the bin corresponding to the 0 energy exchange. This is believed to be caused by `CorrectTOFAxis` bin index being, naturally, an integer value, corresponding to the time channel with the elastic peak centre, while the output from the fitting of the elastic peak can be in principle a floating point, or simply: elastic peak centre falls between two measured time channels.

**Describe the solution you'd like**

To remedy the imperfect correction of the elastic peak position, an offset could be introduced corresponds to fractional bin sizes for better reproduction of fitted elastic peak centre position.

**Additional context**
The offset is visible in the screenshot below.
fitOffQ2_85

### PR DESCRIPTION
**Pull request by darominski**
**Description of work.**

This PR implements a modification of the direct geometry workflow to improve the precision of the elastic peak calibration. Previously, only the elastic peak time channel was communicated with the CorrectTOFAxis algorithm, which limits the precision of the calibration to the time channel with. In result, the elastic peak centres were observed to not fall at 0 energy exchange but around 0.1 meV away from it.

A solution proposed in this PR adds a possiblity to specify a fractional offset that allows to profit from the full information about the peak centre position obtained from elastic peak fitting. The result was initially validated with Stephane Rols and Bjorn Fak, and the proposed solution was found superior to the one currently available in the code.

**To test:**

1. Code review, tests should be passing.
2. Get the attached file and run the following script in the workbench:

```
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import directtools as dt

elp = CreateSingleValuedWorkspace(DataValue=103.42)

DirectILLCollectData( 
        Run = '026079',
        IncidentEnergyCalibration='Energy Calibration ON',
        ElasticChannelWorkspace=elp,  
        OutputWorkspace = 'test'
)
DirectILLReduction(
            InputWorkspace='test',
            OutputWorkspace="out"
)
Q= 2.85
dQ= 0.35 # this means Q in [2.5,3.2]
plot= "out"
fig, axes, cuts = dt.plotconstQ(plot,Q,dQ)
```
The maximum of the plotted distribution should be now very close to 0.0, while when the data value of the ELP is changed to the nearest integer, the peak would fall at about 0.1 meV.

Fixes #33222.
[026079.nxs.gz](https://github.com/mantidproject/mantid/files/7830083/026079.nxs.gz)

Release note added.

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/src/CorrectTOFAxis.cpp [hunk 1]
```diff
@@ -338,13 +339,18 @@ void CorrectTOFAxis::useReferenceWorkspace(const API::MatrixWorkspace_sptr &outp
 void CorrectTOFAxis::correctManually(const API::MatrixWorkspace_sptr &outputWs) {
   const auto &spectrumInfo = m_inputWs->spectrumInfo();
   const double l1 = spectrumInfo.l1();
-  double l2 = 0;
-  double epp = 0;
+  auto l2 = 0.0;
+  auto epp = 0.0;
+  auto const elasticBinIndexOffset =
+      static_cast<double>(getProperty(PropertyNames::ELASTIC_BIN_INDEX)) - static_cast<int>(m_elasticBinIndex);
```
Comment (gui-co) [inline Framework/Algorithms/src/CorrectTOFAxis.cpp:345]: Here, `m_elasticBinIndex` is a conversion in workspace index of the integer part of the property `PropertyNames::ELASTIC_BIN_INDEX`. Can you still do this operation?

**Review by gui-co — COMMENTED** [review]
The test your are providing is working properly. I added few comments in the code.

File: Framework/Algorithms/src/CorrectTOFAxis.cpp [hunk 2]
```diff
@@ -338,13 +339,18 @@ void CorrectTOFAxis::useReferenceWorkspace(const API::MatrixWorkspace_sptr &outp
...
+  auto eppOffset = 0.0;
   g_log.information() << "EPP: " << epp << ".\n";
```
[the remark this answers is not in the document]
Comment (darominski) [reply Framework/Algorithms/src/CorrectTOFAxis.cpp:347]: You are absolutely correct. This has been present in the code before this PR. I am going to move this log print after the epp is set.

**Comment by darominski** [comment]
> The test your are providing is working properly. I added few comments in the code.

I have responded to your comments and taken them into account. Could you have another look at this?

**Review by gui-co — APPROVED** [review]
Thanks for the changes.

**Pull request merged**